controllers/component_build_controller.go

```diff
@@ -75,35 +75,78 @@ const (
 )
 
 var (
-	simpleBuildPipelineCreationTimeMetric       prometheus.Histogram
-	pipelinesAsCodeComponentProvisionTimeMetric prometheus.Histogram
+	initialComponentCreationBuildOrPaCProvisionTimeMetric prometheus.Histogram
+	simpleBuildPipelineCreationTimeMetric                 prometheus.Histogram
+	pipelinesAsCodeComponentProvisionTimeMetric           prometheus.Histogram
+	pipelinesAsCodeComponentUnconfigureTimeMetric         prometheus.Histogram
+	pipelinesAsCodeComponentTriggerTimeMetric             prometheus.Histogram
+	componentTimesForMetrics                                  = map[string]componentMetricsInfo{}
```

Comment:
Any reason to use this mapping? In what cases multiple components are recorded?

Reply:
we have to record all components, because build can have transient error on which it will run another reconcile, so we have to count that as 1 run